### PR TITLE
Add metadata macros

### PR DESCRIPTION
**Pull request by pdeffebach**
Description of API to come

### CODE REVIEW

**Comment by pdeffebach** [comment]
@bkamins 

This is ready for a review. 

The API is very minimal. However I do add two functions, `printlabels` and `printnotes` because working with metadata in data cleaning is pretty frustrating without them. 

Maybe we can have `printlabels` and `printnotes` live in DataFrames.jl?

File: src/metadata.jl [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+"""
+    label!(df, args...)
+
+Assign labels to columns in a data frame using `:col = label` syntax.
+Shorthand for `label!(df, ...)` from TablesMetaDataTools.jl.
+
+```julia-repl
+julia> df = DataFrame(wage = 12);
+
+julia> @label! df :wage = "Wage per hour (USD)";
```
Comment (bkamins) [inline src/metadata.jl:41]: maybe add an example showing how to add more than one label at once?

File: src/metadata.jl [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    Long discussion of variable construction.
+     ";
+
+julia> printnotes(df)
+Column: wage
+────────────
+Long discussion of variable construction.
+```
+
+Unlike labels, notes are appended.
```
Comment (bkamins) [inline src/metadata.jl:122]: I agree here, but then maybe add a comment how to remove a note from a column?

**Comment by bkamins** [comment]
Thank you. Looks good. I would leave `printlabels` and `printnotes` in DataFramesMeta.jl for now as they are only printing things (not returning a value).

**Comment by pdeffebach** [comment]
@bkamins 

I beefed up the printing a little, trying to make it as useful as possible. 

We can change the API at 1.0 if people don't like it. But I think it's pretty useful. 

This is ready for merging. 

API of printing:

# Labels

    printlabels(df, [cols=All()]; unlabelled = true)

Pretty-print all labels in a data frame.

## Arguments

* `cols`: Optional argument to select columns to print. Can
  be any valid multi-column selector, such as `Not(...)`,
  `Between(...)`, or a regular expression.

* `unlabelled`: Keyword argument for whether to print
  the columns without user-defined labels. Deftaults to `true`.
  For column `col` without a user-defined label, `label(df, col)` returns
  the name of the column, `col`.

## Examples
```julia-repl
julia> df = DataFrame(wage = [12], age = [23]);

julia> @label! df :wage = "Hourly wage (2015 USD)";

julia> printlabels(df)
┌────────┬────────────────────────┐
│ Column │                  Label │
├────────┼────────────────────────┤
│   wage │ Hourly wage (2015 USD) │
│    age │                    age │
└────────┴────────────────────────┘

julia> printlabels(df, :wage)
┌────────┬────────────────────────┐
│ Column │                  Label │
├────────┼────────────────────────┤
│   wage │ Hourly wage (2015 USD) │
└────────┴────────────────────────┘

julia> printlabels(df; unlabelled = false)
┌────────┬────────────────────────┐
│ Column │                  Label │
├────────┼────────────────────────┤
│   wage │ Hourly wage (2015 USD) │
└────────┴────────────────────────┘

julia> printlabels(df, r"^wage")
┌────────┬────────────────────────┐
│ Column │                  Label │
├────────┼────────────────────────┤
│   wage │ Hourly wage (2015 USD) │
└────────┴────────────────────────┘
```

# Notes

    printnotes(df, cols = All(); unnoted = false)

Print the notes and labels in a data frame.

## Arguments
* `cols`: Optional argument to select columns to print. Can
  be any valid multi-column selector, such as `Not(...)`,
  `Between(...)`, or a regular expression.
* `unnoted`: Keyword argument for whether to print
  the columns without user-defined notes or labels.

For the purposes of printing, column labels are printed in
addition to notes. However column labels are not returned by
`note(df, col)`.

```
julia> df = DataFrame(wage = [12], age = [23]);

julia> @label! df :age = "Age (years)";

julia> @note! df :wage = "Derived from American Community Survey";

julia> @note! df :wage = "Missing values imputed as 0 wage";

julia> @label! df :wage = "Hourly wage (2015 USD)";

julia> printnotes(df)
Column: wage
────────────
Label: Hourly wage (2015 USD)
Derived from American Community Survey
Missing values imputed as 0 wage

Column: age
───────────
Label: Age (years)
```

**Review by bkamins — APPROVED** [review]
Looks good. Just merge conflicts need to be resolved.

**Comment by pdeffebach** [comment]
Thanks! Will merge!

**Pull request merged**